test/lit/passes/remove-unused-module-elements-refs.wast

```diff
@@ -1,14 +1,26 @@
 ;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
-;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements --nominal -all -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements --closed-world --nominal -all -S -o - | filecheck %s
+;; RUN: foreach %s %t wasm-opt --remove-unused-module-elements                --nominal -all -S -o - | filecheck %s --check-prefix OPEN_WORLD
+
+;; Test both open world (default) and closed world. In a closed world we can do
+;; more with function refs, as we assume nothing calls them on the outside, so
+;; if no calls exist to the right type, the function is not reached.
```

Comment:
Open world does the same as closed here, except it fails to turn some function bodies to unreachable (when closed world can see they are not called, open can't infer that).

Reply:
Which functions? Can we specify them in comments to make this easier to read?

Reply:
I added comments in the tests now, good idea.

Also when reading them carefully I noticed we introduced a bug in a test recently... we had a null for a call_ref input when it should have been a local.get, so it was not testing the right thing. I changed it to match the comments around it.

Reply:
(that is the last commit)